torch/_functorch/_aot_autograd/runtime_wrappers.py

```diff
@@ -1968,6 +1982,16 @@ def call_compiled_backward():
                         steal_args=True,
                         disable_amp=disable_amp,
                     )
+
+                    # Toss out the backward output tokens
+                    num_bw_tokens = CompiledFunction.metadata.num_backward_tokens
+                    if num_bw_tokens > 0:
+                        out_tokens = out[-num_bw_tokens:]
+                        for t in out_tokens:
+                            assert t.numel() == 0
```

Comment:
Just a thought: one (annoying) thing we need to balance in AOTDispatcher at runtime is having enough useful asserts without hurting the hot-path runtime performance too much (which you might see as part of https://github.com/pytorch/pytorch/issues/129457)

`out = out[:-num_bw_tokens]` will materialize a fresh list, so this will be more expensive on backward graphs that have a large number of outputs (empirically, we have internal models with > 100 graph inputs/outputs).

We are doing similarly egregious things in other parts of the runtime wrapper, so I'm ok with leaving this list slicing here for now, but definitely something to think about.

Reply:
Thanks for catching, agree, will remove.